### PR TITLE
[stdlib] Build the standard library as Swift 5

### PR DESCRIPTION
**Pull request by airspeedswift**
4.2 was a non-event so figured ¯\\_(ツ)_/¯ 

Not much needed changing over #17580, mostly warnings:
 - I've disabled deprecating the top-level `Unicode.*` typealiases, for a later patch
 - put a weirdly recursive slow-path handler for unknown `FloatingPointRoundingRule`
 - ~~removed `@inlinable` from debugger support – shouldn't be necessary~~ done in #18126 
 - made `EndodingError` and `DecodingError` resilient.

@itaiferber are you ok with making those two errors resilient? Seems like they'll only occur on slow paths so the perf hit shouldn't matter, and allows for future changes.

@slavapestov making them resilient causes `deserialize_coregraphics.swift` to fail:

```
SIL verification failed: Load must have a loadable type: !fnConv.useLoweredAddresses() || LI->getType().isLoadable(LI->getModule())
Verifying instruction:
     %31 = tuple_element_addr %28 : $*(@thick Any.Type, DecodingError.Context), 1 // user: %32
->   %32 = load %31 : $*DecodingError.Context     // users: %86, %81, %76, %58
     %58 = tuple (%30 : $@thick Any.Type, %32 : $DecodingError.Context) // user: %59
     %76 = struct_extract %32 : $DecodingError.Context, #DecodingError.Context.codingPath // user: %77
     %81 = struct_extract %32 : $DecodingError.Context, #DecodingError.Context.debugDescription // user: %82
     %86 = struct_extract %32 : $DecodingError.Context, #DecodingError.Context.underlyingError // user: %87
In function:
// protocol witness for Decodable.init(from:) in conformance CGFloat
```

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
@swift-ci please test source compatibility

File: cmake/modules/SwiftSource.cmake [hunk 1]
```diff
@@ -259,7 +259,7 @@ function(_compile_swift_files
 
   # Force swift 4 compatibility mode for Standard Library.
```
Comment (jrose-apple) [inline cmake/modules/SwiftSource.cmake]: Probably want to update the comment too.

**Review by stephentyrone — APPROVED** [review]
That's pretty weird, but no, yeah.

**Comment by airspeedswift** [comment]
@slavapestov benchmarks don't look harmed

**Comment by slavapestov** [comment]
Thanks for the crash report, I’ll take a look!

**Review by itaiferber — APPROVED** [review]
LGTM — I believe we added the fixed-layout annotations to `EncodingError` and `DecodingError` back in https://github.com/airspeedswift/swift/commit/a24998a5b12151e8acfbe5f1f70dcf80685334b1; I think it's reasonable to remove them now.

**Comment by itaiferber** [comment]
@airspeedswift Should we remove any other `@_fixed_layout` annotations on the `Codable` stuff to allow for future evolution? AFAICT, it should be fine to leave the other `@_fixed_layout` annotations on other types in that file as none of them should be introducing direct storage, but I just want to confirm.

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@itaiferber Hmm... possibly, but it's pretty generic so it could take quite a hit. How much benchmarking coverage does it have?

**Comment by airspeedswift** [comment]
Gah. Works on my machine™️.

**Comment by airspeedswift** [comment]
@swift-ci please test macOS platform

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
@swift-ci please smoke test compiler performance

**Comment by airspeedswift** [comment]
@swift-ci please test

**Pull request merged**